### PR TITLE
connect events subscription

### PR DESCRIPTION
**Pull request by codev0**
Fixes # .

- fix typos and docs in api
- add events subscriptions

@reviewer-or-team

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@gear-js/api",
-  "version": "0.4.5",
```
Comment (osipov-mit) [inline api/package.json:3]: Why did you downgrade the version?

**Review by osipov-mit — CHANGES_REQUESTED** [review]

**Review by osipov-mit — APPROVED** [review]

File: website/frontend/src/store/actions/actions.ts [hunk 2]
```diff
@@ -244,14 +240,24 @@ export const handleProgramSuccess = () => (dispatch: any, getState: any) => {
...
-  dispatch(fetchNotificationsAction());
-  notificationService
-    .fetchAllNotifications(params)
-    .then((result: NotificationRPCModel) => {
-      dispatch(fetchNotificationsSuccessAction(result.result));
-    })
-    .catch(() => dispatch(fetchNotificationsErrorAction()));
+export const getNotificationsAction = () => (dispatch: any) => {
+  nodeApi.subscribeProgramEvents((event) => {
+    event.data.forEach((i) => {
```
Comment (osipov-mit) [inline website/frontend/src/store/actions/actions.ts:245]:
```javascript
import {  InitSuccessData } from '@gear-js/api'
event.data.forEach((data: InitSuccessData) => {
  if (data.origin.toHex() === localStorage.getItem('public_key_raw')){
    // to do smth
}
})

```

File: website/frontend/src/store/actions/actions.ts [hunk 3]
```diff
@@ -244,14 +240,24 @@ export const handleProgramSuccess = () => (dispatch: any, getState: any) => {
...
+      const data = i.toHuman() as { source: string };
+      if (data.source === localStorage.getItem('public_key_raw')) {
```
Comment (osipov-mit) [inline website/frontend/src/store/actions/actions.ts:247]: `source` property is not exists here. you should use `origin`

File: website/frontend/src/store/actions/actions.ts [hunk 4]
```diff
@@ -244,14 +240,24 @@ export const handleProgramSuccess = () => (dispatch: any, getState: any) => {
...
+        dispatch(fetchNotificationsSuccessAction(data));
+      }
+    });
+  });
+
+  nodeApi.subscribeLogEvents((event) => {
+    event.data.forEach((i) => {
+      const data = i.toHuman() as { source: string };
+      if (data.source === localStorage.getItem('public_key_raw')) {
```
Comment (osipov-mit) [inline website/frontend/src/store/actions/actions.ts:256]: Here `source` means program id, and you should use `dest` property

File: website/frontend/src/store/actions/actions.ts [hunk 5]
```diff
@@ -244,14 +240,24 @@ export const handleProgramSuccess = () => (dispatch: any, getState: any) => {
...
+        dispatch(fetchNotificationsSuccessAction(data));
+      }
+    });
+  });
+
+  nodeApi.subscribeLogEvents((event) => {
+    event.data.forEach((i) => {
```
Comment (osipov-mit) [inline website/frontend/src/store/actions/actions.ts:254]:
```javascript
import {  LogData } from '@gear-js/api'
event.data.forEach((data: LogData) => {
  if (data.dest.toHex() === localStorage.getItem('public_key_raw')){
    // to do smth
}
})
```

File: website/frontend/src/store/actions/actions.ts [hunk 6]
```diff
@@ -244,14 +240,24 @@ export const handleProgramSuccess = () => (dispatch: any, getState: any) => {
...
-export const getNotificationsAction = (params: PaginationModel) => (dispatch: any) => {
-  dispatch(fetchNotificationsAction());
-  notificationService
-    .fetchAllNotifications(params)
-    .then((result: NotificationRPCModel) => {
-      dispatch(fetchNotificationsSuccessAction(result.result));
-    })
-    .catch(() => dispatch(fetchNotificationsErrorAction()));
+export const getNotificationsAction = () => (dispatch: any) => {
+  nodeApi.subscribeProgramEvents((event) => {
```
Comment (osipov-mit) [inline website/frontend/src/store/actions/actions.ts:244]: Here you need to check what event is (InitSuccess or InitFailure)

You can use
```javascript
nodeApi.subscribeProgramEvents(({ data, method }) => {
  // method is event name
  // to do stmh
})
```

**Comment by osipov-mit** [comment]
@codev0 You should merge branch move-node-conn-to-front and install packages with updated @gear-js/api

File: website/frontend/src/store/actions/actions.ts [hunk 7]
```diff
@@ -264,6 +267,49 @@ export const getUnreadNotificationsCount = () => (dispatch: any) => {
...
+            }),
+          })
+        );
+      }
+    });
+  });
+
+  nodeApi.subscribeLogEvents(({ data }) => {
+    // @ts-ignore
+    data.forEach((eventData: LogData) => {
```
Comment (osipov-mit) [inline website/frontend/src/store/actions/actions.ts:298]: Here you should filter events and display only events for current account
```javascript
if (eventData.dest.toHex() === localStorage.getItem('public_key_raw')) 
```

**Review by osipov-mit — CHANGES_REQUESTED** [review]

**Review by EugenWay — APPROVED** [review]

**Pull request merged**